Signal/src/ViewControllers/DebugUI/DebugUIDiskUsage.m

```diff
@@ -0,0 +1,299 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "DebugUIDiskUsage.h"
+#import "OWSTableViewController.h"
+#import "Signal-Swift.h"
+#import <SignalServiceKit/TSDatabaseView.h>
+#import <SignalServiceKit/TSInteraction.h>
+#import <SignalServiceKit/TSStorageManager.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation DebugUIDiskUsage
+
+#pragma mark - Logging
+
++ (NSString *)tag
+{
+    return [NSString stringWithFormat:@"[%@]", self.class];
+}
+
+- (NSString *)tag
+{
+    return self.class.tag;
+}
+
+#pragma mark - Factory Methods
+
+- (NSString *)name
+{
+    return @"Orphans & Disk Usage";
+}
+
+- (nullable OWSTableSection *)sectionForThread:(nullable TSThread *)thread
+{
+    return [OWSTableSection sectionWithTitle:self.name
+                                       items:@[
+                                           [OWSTableItem itemWithTitle:@"Audit & Log"
+                                                           actionBlock:^{
+                                                               [DebugUIDiskUsage auditWithoutCleanup];
```

Comment:
You may find some missing attachment files. I think all of mine correspond to attachments made during development of the "arbitrary attachments" and "oversize text attachments" work.  Please send me the output of this command on your primary.